test/framework/src/main/java/org/elasticsearch/repositories/blobstore/ESBlobStoreRepositoryIntegTestCase.java

```diff
@@ -47,42 +48,44 @@
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertHitCount;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.greaterThan;
+import static org.hamcrest.Matchers.instanceOf;
 import static org.hamcrest.Matchers.is;
 import static org.hamcrest.Matchers.notNullValue;
 import static org.hamcrest.Matchers.nullValue;
 
 /**
- * Basic integration tests for blob-based repository validation.
+ * Integration tests for {@link BlobStoreRepository} implementations.
  */
 public abstract class ESBlobStoreRepositoryIntegTestCase extends ESIntegTestCase {
 
-    protected abstract void createTestRepository(String name, boolean verify);
-
-    protected void afterCreationCheck(Repository repository) {
+    protected abstract String repositoryType();
 
+    protected Settings repositorySettings() {
+        return Settings.EMPTY;
     }
 
-    protected void createAndCheckTestRepository(String name) {
+    protected final String createRepository(final String name) {
         final boolean verify = randomBoolean();
-        createTestRepository(name, verify);
-
-        final Iterable<RepositoriesService> repositoriesServices =
-            internalCluster().getDataOrMasterNodeInstances(RepositoriesService.class);
 
-        for (RepositoriesService repositoriesService : repositoriesServices) {
-            final BlobStoreRepository repository = (BlobStoreRepository) repositoriesService.repository(name);
-
-            afterCreationCheck(repository);
-            assertThat("blob store has to be lazy initialized",
-                repository.getBlobStore(), verify ? is(notNullValue()) : is(nullValue()));
-        }
+        logger.debug("-->  creating repository [name: {}, verify: {}]", name, verify);
+        assertAcked(client().admin().cluster().preparePutRepository(name)
+            .setType(repositoryType())
+            .setVerify(verify)
+            .setSettings(Settings.builder().put(repositorySettings())));
```

Comment:
NIT: I don't think you need the builder here, there's a version of `setSettings` that just takes `Settings` ?

Reply:
Right, thanks